src/cascadia/TerminalControl/SearchBoxControl.xaml

```diff
@@ -0,0 +1,122 @@
+            <TextBox  x:Name="TextBox" AutomationProperties.Name="Search Text" PlaceholderText="Search"
+                KeyDown = "QuerySubmitted"
+                Width="160" Margin="0,0,1,0" HorizontalAlignment="Left" VerticalAlignment="Center">
+            </TextBox>
+
+            <ToggleButton x:Name="SetGoBackwardButton" AutomationProperties.Name="Set Search Backward" HorizontalAlignment="Right"
+                          Padding="1" Style="{ThemeResource ToggleButtonStyle}" Checked="_GoBackwardChecked" IsChecked="True">
+                <FontIcon FontFamily="Segoe MDL2 Assets" Glyph="&#xE74A;" Style="{ThemeResource FontIconStyle}"/>
+            </ToggleButton>
+            <ToggleButton x:Name="SetGoForwardButton" AutomationProperties.Name="Set Search Forward" Padding="1"
+                          Style="{ThemeResource ToggleButtonStyle}" Checked="_GoForwardChecked">
+                <FontIcon FontFamily="Segoe MDL2 Assets" Glyph="&#xE74B;" Style="{ThemeResource FontIconStyle}"/>
+            </ToggleButton>
+        </StackPanel>
+
+        <StackPanel Orientation="Horizontal" HorizontalAlignment="Right">
+            <CheckBox x:Name="CaseSensitivityCheckBox" AutomationProperties.Name="CaseSensitivity" Margin="0,0,2,0"
+                      Checked="_CaseSensitivityChecked" Unchecked="_CaseSensitivityUnChecked"
+                      KeyDown="_CheckboxKeyDown">
+                <CheckBox.Resources>
+                    <ResourceDictionary>
+                        <ResourceDictionary.ThemeDictionaries>
+                            <ResourceDictionary x:Key="Light">
+                                <SolidColorBrush x:Key="CheckBoxColor" Color="White"/>
+                                <StaticResource x:Key="CheckBoxCheckBackgroundStrokeUnchecked" ResourceKey="CheckBoxColor" />
+                                <StaticResource x:Key="CheckBoxCheckBackgroundStrokeUncheckedPointerOver" ResourceKey="CheckBoxColor" />
+                            </ResourceDictionary>
+                            <ResourceDictionary x:Key="Default">
+                                <SolidColorBrush x:Key="CheckBoxColor" Color="White"/>
+                                <StaticResource x:Key="CheckBoxCheckBackgroundStrokeUnchecked" ResourceKey="CheckBoxColor" />
+                                <StaticResource x:Key="CheckBoxCheckBackgroundStrokeUncheckedPointerOver" ResourceKey="CheckBoxColor" />
+                            </ResourceDictionary>
+                            <ResourceDictionary x:Key="HighContrast">
+                                <SolidColorBrush x:Key="CheckBoxColor" Color="{ThemeResource SystemColorButtonTextColor}"/>
+                                <StaticResource x:Key="CheckBoxCheckBackgroundStrokeUnchecked" ResourceKey="CheckBoxColor" />
+                                <StaticResource x:Key="CheckBoxCheckBackgroundStrokeUncheckedPointerOver" ResourceKey="CheckBoxColor" />
+                            </ResourceDictionary>

```

Comment:
I don't know xaml too well but a lot of this code is repeated. Is there any way to cut down on that?

Reply:
I don't know if there is a way to simplify this. I will ask seniors. But actually MinMaxCloseControl.xaml is also doing this.

Reply:
Because that CheckBoxColor's value is distinct in HighContrast mode, I think we can not simplify this.